src/wp-includes/class-wp-block-type.php

```diff
@@ -327,6 +327,17 @@ public function set_props( $args ) {
 
 		$args['name'] = $this->name;
 
+		// Make sure core blocks register a stylesheet.
+		if ( 0 === strpos( $args['name'], 'core/' ) ) {
+			$block_name           = str_replace( 'core/', '', $args['name'] );
+			$args['style']        = isset( $args['style'] )
+				? $args['style']
+				: "wp-block-$block_name";
```

Comment:
Can't you use `generate_block_asset_handle` to get the block handle?

Reply:
Yes, we can use that... But what would the benefit be? :thinking: 
We only want this for core blocks, so the `strpos()` check will need to be there. And if we already check for that, then calling `generate_block_asset_handle` would only run the same check again, and apply logic that won't do anything more than what the above code does.
If we wanted this to also apply for 3rd-party blocks and not be limited to core blocks, then using `generate_block_asset_handle` would make more sense 'cause we'd be able to also get rid of the wrapping `if ( 0 === strpos( $args['name'], 'core/' ) ) {` condition.
But doing this for 3rd-party blocks may be dangerous... We don't know if, how, and when blocks from plugins register their styles, so if we register them here, the plugin ones may break :grimacing:

Reply:
Makes sense. The benefint would be that you ensure same handle, but that can be unit tested.